### PR TITLE
[Feature] Add debate types

### PR DESCRIPTION
**Pull request by jamiebrown201**
Here im implementing adding debate types to the project being populated from a DB. This pull request includes:

- Adding debate type routes
- Splitting out debate from the admin routes to separate debate types and debates.
- Adding the debate type db methods.
- Temporarily disabling the nice client side validation (this can be re added later but I didn't want to spend longer on this).
- Adding server side validation for debate and debate types
- Adding custom and fields that the user can add for special users in debate types
- Adding custom fields where user names can be added to special users (Including the default of blocked and allowed)
- Adding very basic error validation which is hacky but will be hardly used with client side.

At some point in the next couple of weeks im going to add a build process with babel as I have used modern Javascript on the front end because it cut down the lines in the code significantly. This will also be good to have anyway and shouldnt take too long.

You will need to add this to your env
`DEBATE_TYPE_TABLE=ftlabs-order_debate-type`

### CODE REVIEW

File: routes/admin/debate.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    const spcialUsersFormatted = specialUsers
+      ? formatSpecialUsers(specialUsers)
+      : [];
+
+    const params = {
+      title,
+      description,
+      debateStatus,
+      votingStatus,
+      specialUsers: spcialUsersFormatted,
```
Comment (Sqyphen) [inline routes/admin/debate.js]: Should this be _specialUsersFormatted_?

File: routes/admin/debate.js [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+  } catch (err) {
+    console.error(err);
+    res.redirect(
+      `/admin/debate/edit/${debateId}?alertType=error&alertAction=editing`,
+    );
+  }
+});
+
+function formatSpecialUsers(specialUsers) {
+  let spcialUsersFormatted = [];
```
Comment (Sqyphen) [inline routes/admin/debate.js]: Should this be _ specialUsersFormatted_ ? Just missing the _e_ ?

File: routes/api.js [hunk 3]
```diff
@@ -1,71 +1,8 @@
...
-
-    return res.send(JSON.stringify(debate));
-  } catch (err) {
-    return res
-      .status(404)
-      .send("Sorry can't find that! Issue with POST /debate/create");
-  }
-});
-
 router.post('/debate/edit', async (req, res) => {
```
Comment (Sqyphen) [inline routes/api.js]: After moving the debate/edit functions to routes/admin/debate.js, is this file needed anymore?

Comment (jamiebrown201) [reply routes/api.js]: Dont think so

File: routes/admin/debate.js (hunk 1 above)
Comment (jamiebrown201) [reply routes/admin/debate.js]: lol yes!

File: static/css/admin/debateForm.css [hunk 4]
```diff
@@ -0,0 +1,9 @@
+.user-inputs {
```
Comment (Lily2point0) [inline static/css/admin/debateForm.css]: Is there a scss file for that?

File: routes/router.js [hunk 5]
```diff
@@ -5,8 +5,7 @@ const express = require('express');
 const router = express.Router();
 const s3o = require('@financial-times/s3o-middleware');
 const path = require('path');
-const apiRoutes = require('../routes/api');
-const adminRoutes = require('../routes/admin');
+const adminRoutes = require('./admin/main');
```
Comment (Lily2point0) [inline routes/router.js]: I get `apiRoutes` is not defined further down in that file. At `router.use('/api', apiRoutes);`
Is that still needed?

File: views/admin/editDebate.hbs [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+								<span>&#8722;</span>
+								<span class="o-buttons-icon__label">Remove</span>
+							</button>
+						</div>
+						<div id="msg_error" class="error o-forms__errortext hide" data-validation="{{userType}}"></div>
+					</div>
+					{{/each}}
+				</div>
+				<div class="o-forms">
+					<button type="submit" class="o-buttons o-buttons--primary" id="submit_new_debate">Edit new
```
Comment (Sqyphen) [inline views/admin/editDebate.hbs]: Too much spacing here on _Edit new debate_
Is this prettier trying to help?

Comment (Sqyphen) [reply views/admin/editDebate.hbs]: Sorry, it is Prettier 😐

File: static/css/admin/debateForm.css (hunk 4 above)
Comment (jamiebrown201) [reply static/css/admin/debateForm.css]: No sorry I forgot we had moved to sass. Should I move this to a sass file?
